sklearn/pipeline.py

```diff
@@ -177,20 +177,31 @@ def _validate_steps(self):
         for t in transformers:
             if t is None or t == 'passthrough':
                 continue
-            if (not (hasattr(t, "fit") or hasattr(t, "fit_transform")) or not
-                    hasattr(t, "transform")):
+            if (not hasattr(t, "fit_transform") and
+                    (not (hasattr(t, "fit") and hasattr(t, "transform")))):
                 raise TypeError("All intermediate steps should be "
-                                "transformers and implement fit and transform "
-                                "or be the string 'passthrough' "
-                                "'%s' (type %s) doesn't" % (t, type(t)))
+                                "transformers and implement fit and "
+                                "transform, fit_transform or be the string "
+                                "'passthrough'. '%s' (type %s) doesn't"
+                                % (t, type(t)))
+            elif not hasattr(t, "transform"):
+                warnings.warn("Intermediate step '%s' (type %s) does not have "
+                              "transform, pipeline is not reusable."
+                              % (t, type(t)))
 
         # We allow last estimator to be None as an identity transformation
         if (estimator is not None and estimator != 'passthrough'
-                and not hasattr(estimator, "fit")):
+                and not (hasattr(estimator, "fit") or
+                         hasattr(estimator, "fit_transform"))):
```

Comment:
I don't think we have or allow estimators that support fit_transform but not fit.

Reply:
Okay, I removed that check